File: packages/elixir-sdk/test/adapters/storage/blobs_test.exs
defmodule TreeDxSdk.BlobsAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Blobs.read(client, "repo/a", %{})
    TreeDxSdk.Blobs.write(client, "ws/a", %{})
    TreeDxSdk.Blobs.download(client, "ws/a")
    TreeDxSdk.Blobs.upload(client, "ws/a", "x")
    TreeDxSdk.Blobs.create_multipart_upload(client, "ws/a", %{})
    TreeDxSdk.Blobs.upload_part(client, "ws/a", "up/a", 2, "x")
    TreeDxSdk.Blobs.complete_multipart_upload(client, "ws/a", "up/a", %{})
    TreeDxSdk.Blobs.abort_multipart_upload(client, "ws/a", "up/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :delete and
                 &1.path == "/api/v1/workspaces/ws%2Fa/blobs/uploads/up%2Fa")
           )
  end
end

File: packages/ts-sdk/test/adapters/storage/blobs.test.ts
import { describe, expect, it } from 'vitest';
import { BlobsAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('BlobsAdapter', () => {
  it('constructs blob and multipart endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new BlobsAdapter({ transport });
    await adapter.read('repo', {});
    await adapter.write('ws', {});
    await adapter.download('ws');
    await adapter.upload('ws', new Uint8Array([1]));
    await adapter.createMultipartUpload('ws', {});
    await adapter.uploadPart('ws', 'up', 1, new Uint8Array([1]));
    await adapter.completeMultipartUpload('ws', 'up', {});
    await adapter.abortMultipartUpload('ws', 'up');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/blobs/read',
      'POST /api/v1/workspaces/ws/blobs/write',
      'GET /api/v1/workspaces/ws/blobs/download',
      'PUT /api/v1/workspaces/ws/blobs/upload',
      'POST /api/v1/workspaces/ws/blobs/uploads',
      'PUT /api/v1/workspaces/ws/blobs/uploads/up/parts/1',
      'POST /api/v1/workspaces/ws/blobs/uploads/up/complete',
      'DELETE /api/v1/workspaces/ws/blobs/uploads/up'
    ]);
  });
});
